source/postprocess/visualization.cc

```diff
@@ -161,6 +161,38 @@ namespace aspect
           }
       };
 
+      /**
+       * This Postprocessor will generate the output variables of velocity,
+       * pressure, temperature, and compositional fields on the surface of the
+       * domain.
+       */
+      template <int dim>
+      class SurfaceBaseVariablePostprocessor: public VisualizationPostprocessors::SurfaceOnlyVisualization<dim>, public BaseVariablePostprocessor< dim >
+      {
+        public:
+
+          std::vector<std::string> get_names () const override
+          {
+            std::vector<std::string> solution_names (dim, "surface_velocity");
+
+            if (this->include_melt_transport())
+              {
+                solution_names.emplace_back("surface_p_f");
+                solution_names.emplace_back("surface_p_c_bar");
+                for (unsigned int i=0; i<dim; ++i)
+                  solution_names.emplace_back("surface_u_f");
+              }
+            solution_names.emplace_back("surface_p");
+
+            solution_names.emplace_back("surface_T");
+            for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+              solution_names.push_back ("surface_" + this->introspection().name_for_compositional_index(c));
+
+            return solution_names;
+          }
```

Comment:
This duplicates a lot of code from the previous class. Could you do something like the following?
```suggestion
          std::vector<std::string> get_names () const override
          {
            std::vector<std::string> solution_names = BaseVariablePostprocessor<dim>::get_names();
            for (std::string &name : solution_names)
              name = "surface_" + name;
            return solution_names;
          }
```